### ISSUE TITLE
main page hangs, http error

### ISSUE THREAD

**Issue by wrought** [sudoroom/sudo-humans#12]
for `GET` request on `/`, incomplete response due to: `net::ERR_INCOMPLETE_CHUNKED_ENCODING`

I seem to get this issue when I first run the application with `npm start` before there are any users (the main page seems to load forever, so I jump to the "Create Account" page to get out of it).

After I add users, things seem to be fine... until I kill the app process and start it back up again with `npm start`. At this point, the main page might hang as well.

The only resolution I have found is to delete the database and start over, which works, but is a bummer.

**Comment by wrought** [comment]
The other way to get past this problem without deleting the db is to create a new user. If you were already signed when you killed the app process, upon restart you are stuck signed in and cannot sign out, so this requires navigating directly to localhost:8000/account/create

**Comment by ghost** [comment]
What version of node are you using? Does the process crash with any errors?

**Comment by wrought** [comment]
no crash, versions:

`$ npm --version`
`1.3.11`
`$ node --version`
`v0.10.21`

**Comment by wrought** [comment]
although just to be sure, I upgraded using

```
$ sudo npm install -g n
$ sudo n stable
```

now I'm on

```
$ node --version
v0.12.0
```

**Comment by wrought** [comment]
now I get an error, here's the debug log:

```
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/usr/local/bin/npm', 'start' ]
2 info using npm@2.5.0
3 info using node@v0.12.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info prestart sudo-humans@0.0.0
6 info start sudo-humans@0.0.0
7 verbose unsafe-perm in lifecycle true
8 info sudo-humans@0.0.0 Failed to exec start script
9 verbose stack Error: sudo-humans@0.0.0 start: `bin/cmd.js`
9 verbose stack Exit status 1
9 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:213:16)
9 verbose stack     at EventEmitter.emit (events.js:110:17)
9 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:14:12)
9 verbose stack     at ChildProcess.emit (events.js:110:17)
9 verbose stack     at maybeClose (child_process.js:1008:16)
9 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1080:5)
10 verbose pkgid sudo-humans@0.0.0
11 verbose cwd /home/wrought/projects/sudo-humans
12 error Linux 3.2.0-4-amd64
13 error argv "node" "/usr/local/bin/npm" "start"
14 error node v0.12.0
15 error npm  v2.5.0
16 error code ELIFECYCLE
17 error sudo-humans@0.0.0 start: `bin/cmd.js`
17 error Exit status 1
18 error Failed at the sudo-humans@0.0.0 start script 'bin/cmd.js'.
18 error This is most likely a problem with the sudo-humans package,
18 error not with npm itself.
18 error Tell the author that this fails on your system:
18 error     bin/cmd.js
18 error You can get their info via:
18 error     npm owner ls sudo-humans
18 error There is likely additional logging output above.
19 verbose exit [ 1, true ]
```

**Comment by wrought** [comment]
More details:

```
$ npm start

> sudo-humans@0.0.0 start sudo-humans
> bin/cmd.js

sudo-humans/node_modules/tcp-bind/index.js:33
    throw ex;
          ^
Error: bind undefined
    at error (sudo-humans/node_modules/tcp-bind/index.js:30:14)
    at module.exports (sudo-humans/node_modules/tcp-bind/index.js:17:9)
    at Object.<anonymous> (sudo-humans/bin/cmd.js:25:10)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3
```

**Comment by ghost** [comment]
tcp-bind doesn't seem to work on node 0.12.

**Comment by ghost** [comment]
And the `canvas` module hasn't been updated to work with 0.12 yet either. Probably best to stick to 0.10 for now until those modules get fixed upstream.

**Comment by ghost** [comment]
Just fixed tcp-bind in 1.2.0 but the canvas issue remains.
